contracts/instruments/RibbonVolatility.sol

```diff
@@ -436,14 +436,47 @@ contract RibbonVolatility is DSMath, InstrumentStorageV1, InstrumentStorageV2 {
         emit Exercised(msg.sender, positionID, totalProfit, optionsExercised);
     }
 
-    function claimRewards(string calldata adapterName, address rewardsAddress, uint256[] calldata optionIDs)
+    function claimRewards(string calldata adapterName, address rewardsAddress)
         external
     {
       IProtocolAdapter adapter = IProtocolAdapter(factory.getAdapter(adapterName));
+      uint256[] memory optionIDs = getOptionIDs(msg.sender);
       adapter.delegateClaimRewards(rewardsAddress, optionIDs);
       emit ClaimedRewards(optionIDs.length);
     }
 
+    function rewardsClaimable(string calldata adapterName, address rewardsAddress)
+        external
+        view
+        returns (uint256 rewardsToClaim)
+    {
+      IProtocolAdapter adapter = IProtocolAdapter(factory.getAdapter(adapterName));
+      uint256[] memory optionIDs = getOptionIDs(msg.sender);
+      rewardsToClaim = adapter.delegateRewardsClaimable(rewardsAddress, optionIDs);
+    }
+
+    function getOptionIDs(address user)
+        private
+        view
+        returns (uint256[] memory optionIDs)
+    {
+        uint256 i = 0;
+        uint256 j = 0;
+        InstrumentPosition[] memory positions = instrumentPositions[user];
+        optionIDs = new uint256[](positions.length * 2);
```

Comment:
have to make statically sized if I want it in memory. in the case that positions.length * 2 (max optionids possible) is more than is actually filled, we check in hegicadapter if optionids[i] > 0 (0 is initial value in uninitialized uint256)